web-services-app/src/test/java/edu/unc/lib/boxc/web/services/processing/DownloadImageServiceTest.java

```diff
@@ -0,0 +1,254 @@
+                .thenReturn(jp2Datastream);
+        when(jp2Datastream.getExtent()).thenReturn("800x500");
+        assertEquals(FULL_SIZE, downloadImageService.getSize(contentObjectRecord, "800"));
+    }
+
+    @Test
+    public void testGetSize() {
+        when(contentObjectRecord.getDatastreamObject(eq(DatastreamType.JP2_ACCESS_COPY.getId())))
+                .thenReturn(jp2Datastream);
+        when(jp2Datastream.getExtent()).thenReturn("500x500");
+        assertEquals("125", downloadImageService.getSize(contentObjectRecord, "125"));
+    }
+
+    @Test
+    public void testGetSizeOriginalFileDatastream() {
+        when(contentObjectRecord.getDatastreamObject(eq(DatastreamType.JP2_ACCESS_COPY.getId())))
+                .thenReturn(null);
+        when(contentObjectRecord.getDatastreamObject(eq(DatastreamType.ORIGINAL_FILE.getId())))
+                .thenReturn(datastream);
+
+    }
+
+    @Test
+    public void testStreamDownload() throws IOException {
+        var constructorArgs = new ArrayList<>();
+        try (MockedConstruction<UrlResource> urlResourceClass = Mockito.mockConstruction(UrlResource.class,
+                // get the URL sent into UrlResource constructor
+                (mock, context) -> constructorArgs.add(context.arguments().get(0)))){
+            var contentDispositionHeader = "attachment; filename=original_file_max.jpg";
+            var contentType = MediaType.IMAGE_JPEG;
+            var pid = makePid();
+            var url = ImageServerUtil.buildURL(IIIF_BASE, pid.getId(), FULL_SIZE);
+            when(contentObjectRecord.getDatastreamObject(any())).thenReturn(datastream);
+            when(datastream.getFilename()).thenReturn("original_file.png");
+            when(contentObjectRecord.getPid()).thenReturn(pid);
+
+            var downloadStream = downloadImageService.streamDownload(contentObjectRecord, FULL_SIZE);
+            var mockedUrlResource = urlResourceClass.constructed().get(0);
+
+            var expectedResponse = ResponseEntity.ok()
+                    .header(HttpHeaders.CONTENT_DISPOSITION, contentDispositionHeader)
+                    .contentType(contentType)
+                    .body(mockedUrlResource);
+            assertEquals(expectedResponse, downloadStream);
+            assertEquals(url, constructorArgs.get(0));
+        }
+    }
+
+    @Test
+    public void testStreamThumbnail() throws IOException {
+        var constructorArgs = new ArrayList<>();
+        try (MockedConstruction<UrlResource> urlResourceClass = Mockito.mockConstruction(UrlResource.class,
+                // get the URL sent into UrlResource constructor
+                (mock, context) -> constructorArgs.add(context.arguments().get(0))))
+        {
+            var contentDispositionHeader = "inline;";
+            var contentType = MediaType.IMAGE_JPEG;
+            var pid = makePid();
+            var url = ImageServerUtil.buildURL(IIIF_BASE, pid.getId(), FULL_SIZE);
```

Comment:
realistically, a thumbnail request isn't going to be for the full size of the image since that wouldn't really be a thumbnail, although I don't think it makes much difference to this test.